holoviews/streams.py

```diff
@@ -673,12 +673,20 @@ def __init__(self, parameterized=None, parameters=None, watch=True, watch_only=F
         super(Params, self).__init__(parameterized=parameterized, parameters=parameters, **params)
         self._memoize_counter = 0
         self._events = []
+        self._watchers = []
         if watch:
             # Subscribe to parameters
             keyfn = lambda x: id(x.owner)
             for _, group in groupby(sorted(parameters, key=keyfn)):
                 group = list(group)
-                group[0].owner.param.watch(self._watcher, [p.name for p in group])
```

Comment:
You might want to consider making the body of this `if watch` clause a separate `watch` method.